src/promptflow/promptflow/_utils/version_hint_utils.py

```diff
@@ -0,0 +1,117 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+import datetime
+import json
+import sys
+import contextlib
+
+from promptflow._constants import (LAST_HINT_TIME, LAST_CHECK_TIME, PF_VERSION_CHECK, CLI_PACKAGE_NAME,
+                                   HINT_FREQUENCY_DAY, GET_PYPI_FREQUENCY_DAY, LATEST_VERSION, CURRENT_VERSION)
+from promptflow._sdk._constants import HOME_PROMPT_FLOW_DIR
+
+
+if sys.platform.startswith("win"):
```

Comment:
any special reason to use `msvcrt`/`fcntl` as lock pkg? previously, we use locks in python and `filelock` as file lock, so I'd like to hear the reason using another lock here.